drgn/helpers/linux/pid.py

```diff
@@ -123,3 +125,39 @@ def for_each_task_in_group(
     ):
         if other != task or include_self:
             yield other
+
+
+def is_kthread(t: Object) -> bool:
+    """
+    Check if a task is kernel thread.
+    """
+    if not t.mm and not task_state_to_char(t) == "Z":
+        return True
+    return False
+
+
+def is_user(t: Object) -> bool:
+    """
+    Check if a task is user thread.
+    """
+    if t.mm:
+        return True
+    elif task_state_to_char(t) == "Z":
+        return True
+    return False
+
+
+def is_group_leader(t: Object) -> bool:
+    """
+    Check if a task is thread group leader.
+    """
+    if t.pid.value_() == t.tgid.value_():
+        return True
+    return False
+
+
+def get_command(task: Object) -> str:
+    """
+    :returns: name of the command
+    """
+    return escape_ascii_string(task.comm.string_())
```

Comment:
This one is trivial, so I'd rather not add it as a helper. It can go in `ps.py` if it makes your life easier, or you can just fold it into the caller. The `--drgn` output can just suggest `task.comm`.